pylds/base_discrete.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+from operator import itemgetter
+from pylds.base import generate_points, lagrangian_descriptor
+
+def check_if_points_escape_box(u, box_boundaries):
+    x, y = u.T
+    # Escape condition
+    box_x_min, box_x_max = box_boundaries[0]
+    box_y_min, box_y_max = box_boundaries[1]
+    u_indices = (x >= box_x_min) & (x <= box_x_max) & (y >= box_y_min) & (y <= box_y_max)
+    return u_indices
+
+def compute_lagrangian_descriptor(grid_parameters, discrete_map, N_iterations, p_value=0.5, box_boundaries=False):
+    y0, mask = generate_points(grid_parameters)
+    y0 = y0.reshape(-1,3)
+    y0 = y0[:,:-1]
```

Comment:
`Generate_points` returns a flattened array that in the very next step is reverted back to a 2D array. Can't this be avoided? Also the reordering seems like something that should be addressed before the genrating the grid rather than in postprocessing

Reply:
This step cannot be avoided. Either it's done here or in some other function called under `compute_lagrangian_descriptor` that depends on `y0`, such as `check_if_points_escape_box` or `f` (which is the `discrete_map`) which maps `ndarrays`.

In the `base.py` module for continuous systems, the above step was done implicitly under `vector_field_flat` to be able to pass reshaped values to `vector_field` to map `ndarrays`.

Reply:
Redefine generate grid to output nd arrays instead of 1d

Reply:
@vkrajnak I will fix this once the pull-request has been accepted. 

As this change concerns not only discrete systems, I opened an issue (#35 ) for this and modify `generate_grid` to output an `ndarray` and also both versions of `compute_lagrangian_descriptor` in `base.py` and `base_discrete.py` to take this change into account.